### PR TITLE
DA-847 updated the sdk to allow empty profile sharing.

### PR DESCRIPTION
**Pull request by Alttaf**
update to allow empty profiles, also updated package.json to add dependencies  to package.json

### CODE REVIEW

File: src/profile_service/index.js [hunk 1]
```diff
@@ -79,12 +79,17 @@ function getRSASignatureForMessage(message, pem) {
 }
 
 function decryptCurrentUserReceipt(receipt, pem, callback) {
-  let unwrappedKey = unwrapKey(receipt.wrapped_receipt_key, pem);
-  let decodedData = protoRoot.decodeEncryptedData(new Buffer(receipt.other_party_profile_content, 'base64'))
-  let iv = forge.util.decode64(decodedData.iv);
-  let cipherText = forge.util.decode64(decodedData.cipherText);
-
-  return decipherProfile(cipherText, forge.util.decode64(unwrappedKey), iv);
+  if(receipt.other_party_profile_content && Object.keys(receipt.other_party_profile_content).length > 0) {
```
Comment (vassyz) [inline src/profile_service/index.js]: I usually replace the if-else block with guard clause:

```javascript
function decryptCurrentUserReceipt(receipt, pem, callback) {
  if (!receipt.other_party_profile_content || Object.keys(receipt.other_party_profile_content).length === 0) {
    console.log('no decrypted data')
    return []
  }

  const unwrappedKey = unwrapKey(receipt.wrapped_receipt_key, pem);
  const decodedData = protoRoot.decodeEncryptedData(new Buffer(receipt.other_party_profile_content, 'base64'))
  const iv = forge.util.decode64(decodedData.iv);
  const cipherText = forge.util.decode64(decodedData.cipherText);

  return decipherProfile(cipherText, forge.util.decode64(unwrappedKey), iv);
}
```
Works either way, so you can ignore this.

**Comment by vassyz** [comment]
Looks good, there's a lot of `let` statements that should be `const`, especially for test mocks. But it seems the whole SDK was written like this.

Don't forget to bump the package version number.

File: example/simple-login/package.json [hunk 2]
```diff
@@ -7,6 +7,8 @@
   "scripts": {},
   "dependencies": {
     "yoti-node-sdk" : "1.0.0",
-    "ejs": "^2.4.2"
+    "ejs": "^2.4.2",
```
Comment (xchangebit) [inline example/simple-login/package.json]: pin the version please

File: example/simple-login/package.json [hunk 3]
```diff
@@ -7,6 +7,8 @@
...
+    "express":"^4.15.2",
```
Comment (xchangebit) [inline example/simple-login/package.json]: pin the version please

File: example/simple-login/package.json [hunk 4]
```diff
@@ -7,6 +7,8 @@
...
+    "body-parser":"^1.17.1"
```
Comment (xchangebit) [inline example/simple-login/package.json]: pin the version please

**Comment by Alttaf** [comment]
made updates to version number and pinned dependencies thanks guys

**Pull request merged**